### ISSUE TITLE
WSL2: `NotImplementedError: sudo only supported on Linux`

### ISSUE THREAD

**Issue by paxsali** [kdeldycke/meta-package-manager#976]
### What happened?

Hi again,

there's one little issue I found. I believe this is **directly** related to [this](https://github.com/kdeldycke/meta-package-manager/issues/944) issue:

```shell-session
paxsali{WSL:Ubuntu-22.04}@devbox:[~]$ mpm -o plain --apt upgrade
warning: No package provided, assume -A/--all option.
Upgrade all outdated packages from apt...
Fallback to calling upgrade operation on each outdated package.
warning:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
```
```pytb
Traceback (most recent call last):
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/base.py", line 963, in upgrade
    cli = self.upgrade_all_cli()
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/managers/apt.py", line 225, in upgrade_all_cli
    return self.build_cli("--yes", "upgrade", sudo=True)
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/base.py", line 735, in build_cli
    raise NotImplementedError("sudo only supported on Linux.")
NotImplementedError: sudo only supported on Linux.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/paxsali/.local/bin/mpm", line 8, in <module>
    sys.exit(main())
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/__main__.py", line 56, in main
    mpm(prog_name=mpm.name)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click_extra/commands.py", line 159, in main
    return super().main(*args, **kwargs)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click_extra/commands.py", line 213, in invoke
    return super().invoke(ctx)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/paxsali/.local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/cli.py", line 892, in upgrade
    output = manager.upgrade()
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/base.py", line 970, in upgrade
    output = self.upgrade(package.id)
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/base.py", line 959, in upgrade
    cli = self.upgrade_one_cli(package_id, version=version)
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/capabilities.py", line 81, in print_warning
    return func(*args, **kwargs)
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/managers/apt.py", line 238, in upgrade_one_cli
    return self.build_cli(
  File "/home/paxsali/.local/lib/python3.10/site-packages/meta_package_manager/base.py", line 735, in build_cli
    raise NotImplementedError("sudo only supported on Linux.")
NotImplementedError: sudo only supported on Linux.
```

When invoking `mpm` as a regular (non-root) user, obviously for some manager types, like `apt`, you need elevated priviledges for root, like it is usually done via `sudo`.

For some reason the `sudo` functionality in mpm is "guarded" behind one more OS detection check procedure, which makes sense.

Here, again ..., it incorrectly detects the OS and says that `sudo` is only supported on Linux.

Set aside the obvious incorrectness of the wording (what about MacOS or BSD, AIX, etc.? don't they have `sudo`, too?) ..., it should have run on my WSL(2) environment, because it is (essentially) a Linux.

Unfortunatelly above error message doesn't state **what** it was detected as, unlike my initial post from above.

Maybe the test for this is literally a negation? If not OS equals Linux then throw this error? Seems incorrect to me.

Or maybe the idea behind this is more _practically oriented_ than _formally correct_ as in: there are no (not yet) BSD, MacOS, ... "non-Linux" OSes with `apt` as their main pkg-manager (that, I would at least understand).

Does this info help?


### Meta Package Manager configuration

```shell-session
╭────────────────────────────┬────────────────────────────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬──────┬──────────────────┬─────────┬─────────────────────────┬──────────────────────────────────────────────────────────┬──────────────────────────────────────────────────────────┬─────────────╮
│ ID                         │ Class                                  │ Spec.                                                                                                                                                                                                                                                                                                                                                                                                                           │ Type │ Allowed in conf? │ Exposed │ Env. var.               │ Default                                                  │ Value                                                    │ Source      │
├────────────────────────────┼────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼──────┼──────────────────┼─────────┼─────────────────────────┼──────────────────────────────────────────────────────────┼──────────────────────────────────────────────────────────┼─────────────┤
│ mpm.all_managers           │ cloup._params.Option                   │ -a, --all-managers                                                                                                                                                                                                                                                                                                                                                                                                              │ bool │                  │ ✓       │ MPM_ALL_MANAGERS        │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.apm                    │ cloup._params.Option                   │ --apm                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_APM                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.apt                    │ cloup._params.Option                   │ --apt                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_APT                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.apt_mint               │ cloup._params.Option                   │ --apt-mint                                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✘       │ MPM_APT_MINT            │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.bar_plugin_path        │ cloup._params.Option                   │ --bar-plugin-path                                                                                                                                                                                                                                                                                                                                                                                                               │ bool │                  │ ✘       │ MPM_BAR_PLUGIN_PATH     │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.brew                   │ cloup._params.Option                   │ --brew                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_BREW                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.cargo                  │ cloup._params.Option                   │ --cargo                                                                                                                                                                                                                                                                                                                                                                                                                         │ bool │                  │ ✘       │ MPM_CARGO               │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.cask                   │ cloup._params.Option                   │ --cask                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_CASK                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.choco                  │ cloup._params.Option                   │ --choco                                                                                                                                                                                                                                                                                                                                                                                                                         │ bool │                  │ ✘       │ MPM_CHOCO               │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.color                  │ click_extra.colorize.ColorOption       │ --color, --ansi / --no-color, --no-ansi                                                                                                                                                                                                                                                                                                                                                                                         │ bool │                  │ ✘       │ MPM_COLOR               │ True                                                     │ True                                                     │ DEFAULT     │
│ mpm.composer               │ cloup._params.Option                   │ --composer                                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✘       │ MPM_COMPOSER            │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.config                 │ click_extra.config.ConfigOption        │ -C, --config CONFIG_PATH                                                                                                                                                                                                                                                                                                                                                                                                        │ str  │                  │ ✘       │ MPM_CONFIG              │ /home/paxsali/.config/mpm/*.{toml,yaml,yml,json,ini,xml} │ /home/paxsali/.config/mpm/*.{toml,yaml,yml,json,ini,xml} │ DEFAULT     │
│ mpm.description            │ cloup._params.Option                   │ --description                                                                                                                                                                                                                                                                                                                                                                                                                   │ bool │                  │ ✓       │ MPM_DESCRIPTION         │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.dnf                    │ cloup._params.Option                   │ --dnf                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_DNF                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.dry_run                │ cloup._params.Option                   │ -d, --dry-run                                                                                                                                                                                                                                                                                                                                                                                                                   │ bool │                  │ ✓       │ MPM_DRY_RUN             │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.emerge                 │ cloup._params.Option                   │ --emerge                                                                                                                                                                                                                                                                                                                                                                                                                        │ bool │                  │ ✘       │ MPM_EMERGE              │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.exclude                │ cloup._params.Option                   │ -e, --exclude [apm|apt|apt-mint|brew|cargo|cask|choco|composer|dnf|emerge|flatpak|gem|mas|npm|opkg|pacaur|pacman|paru|pip|pipx|scoop|snap|steamcmd|vscode|yarn|yay|yum|zypper]                                                                                                                                                                                                                                                  │ list │                  │ ✓       │ MPM_EXCLUDE             │                                                          │                                                          │ DEFAULT     │
│ mpm.flatpak                │ cloup._params.Option                   │ --flatpak                                                                                                                                                                                                                                                                                                                                                                                                                       │ bool │                  │ ✘       │ MPM_FLATPAK             │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.gem                    │ cloup._params.Option                   │ --gem                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_GEM                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.help                   │ click_extra.colorize.HelpOption        │ -h, --help                                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✘       │ MPM_HELP                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.ignore_auto_updates    │ cloup._params.Option                   │ --ignore-auto-updates / --include-auto-updates                                                                                                                                                                                                                                                                                                                                                                                  │ bool │                  │ ✓       │ MPM_IGNORE_AUTO_UPDATES │ True                                                     │ True                                                     │ DEFAULT     │
│ mpm.manager                │ cloup._params.Option                   │ -m, --manager [apm|apt|apt-mint|brew|cargo|cask|choco|composer|dnf|emerge|flatpak|gem|mas|npm|opkg|pacaur|pacman|paru|pip|pipx|scoop|snap|steamcmd|vscode|yarn|yay|yum|zypper]                                                                                                                                                                                                                                                  │ list │                  │ ✓       │ MPM_MANAGER             │                                                          │                                                          │ DEFAULT     │
│ mpm.mas                    │ cloup._params.Option                   │ --mas                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_MAS                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.npm                    │ cloup._params.Option                   │ --npm                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_NPM                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.opkg                   │ cloup._params.Option                   │ --opkg                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_OPKG                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.output_format          │ click_extra.tabulate.TableFormatOption │ -o, --output-format [asciidoc|csv|csv-excel|csv-excel-tab|csv-unix|double_grid|double_outline|fancy_grid|fancy_outline|github|grid|heavy_grid|heavy_outline|html|jira|json|latex|latex_booktabs|latex_longtable|latex_raw|mediawiki|mixed_grid|mixed_outline|moinmoin|orgtbl|outline|pipe|plain|presto|pretty|psql|rounded_grid|rounded_outline|rst|simple|simple_grid|simple_outline|textile|tsv|unsafehtml|vertical|youtrack] │ str  │                  │ ✘       │ MPM_OUTPUT_FORMAT       │ rounded_outline                                          │ rounded_outline                                          │ DEFAULT     │
│ mpm.pacaur                 │ cloup._params.Option                   │ --pacaur                                                                                                                                                                                                                                                                                                                                                                                                                        │ bool │                  │ ✘       │ MPM_PACAUR              │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.pacman                 │ cloup._params.Option                   │ --pacman                                                                                                                                                                                                                                                                                                                                                                                                                        │ bool │                  │ ✘       │ MPM_PACMAN              │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.paru                   │ cloup._params.Option                   │ --paru                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_PARU                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.pip                    │ cloup._params.Option                   │ --pip                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_PIP                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.pipx                   │ cloup._params.Option                   │ --pipx                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_PIPX                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.scoop                  │ cloup._params.Option                   │ --scoop                                                                                                                                                                                                                                                                                                                                                                                                                         │ bool │                  │ ✘       │ MPM_SCOOP               │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.show_params            │ click_extra.config.ShowParamsOption    │ --show-params                                                                                                                                                                                                                                                                                                                                                                                                                   │ bool │                  │ ✘       │ MPM_SHOW_PARAMS         │ False                                                    │ True                                                     │ COMMANDLINE │
│ mpm.snap                   │ cloup._params.Option                   │ --snap                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_SNAP                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.sort_by                │ cloup._params.Option                   │ -s, --sort-by [manager_id|manager_name|package_id|package_name|version]                                                                                                                                                                                                                                                                                                                                                         │ str  │                  │ ✓       │ MPM_SORT_BY             │ manager_id                                               │ manager_id                                               │ DEFAULT     │
│ mpm.stats                  │ cloup._params.Option                   │ --stats / --no-stats                                                                                                                                                                                                                                                                                                                                                                                                            │ bool │                  │ ✓       │ MPM_STATS               │ True                                                     │ True                                                     │ DEFAULT     │
│ mpm.steamcmd               │ cloup._params.Option                   │ --steamcmd                                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✘       │ MPM_STEAMCMD            │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.stop_on_error          │ cloup._params.Option                   │ --stop-on-error / --continue-on-error                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✓       │ MPM_STOP_ON_ERROR       │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.time                   │ click_extra.commands.TimerOption       │ --time / --no-time                                                                                                                                                                                                                                                                                                                                                                                                              │ bool │                  │ ✘       │ MPM_TIME                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.verbosity              │ click_extra.logging.VerbosityOption    │ -v, --verbosity LEVEL                                                                                                                                                                                                                                                                                                                                                                                                           │ str  │                  │ ✘       │ MPM_VERBOSITY           │ INFO                                                     │ INFO                                                     │ DEFAULT     │
│ mpm.version                │ click_extra.version.VersionOption      │ --version                                                                                                                                                                                                                                                                                                                                                                                                                       │ bool │                  │ ✘       │ MPM_VERSION             │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.vscode                 │ cloup._params.Option                   │ --vscode                                                                                                                                                                                                                                                                                                                                                                                                                        │ bool │                  │ ✘       │ MPM_VSCODE              │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.xkcd                   │ cloup._params.Option                   │ -x, --xkcd                                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✓       │ MPM_XKCD                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.yarn                   │ cloup._params.Option                   │ --yarn                                                                                                                                                                                                                                                                                                                                                                                                                          │ bool │                  │ ✘       │ MPM_YARN                │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.yay                    │ cloup._params.Option                   │ --yay                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_YAY                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.yum                    │ cloup._params.Option                   │ --yum                                                                                                                                                                                                                                                                                                                                                                                                                           │ bool │                  │ ✘       │ MPM_YUM                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.zypper                 │ cloup._params.Option                   │ --zypper                                                                                                                                                                                                                                                                                                                                                                                                                        │ bool │                  │ ✘       │ MPM_ZYPPER              │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.backup.merge           │ cloup._params.Option                   │ --merge                                                                                                                                                                                                                                                                                                                                                                                                                         │ bool │                  │ ✓       │ MPM_MERGE               │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.backup.overwrite       │ cloup._params.Option                   │ --overwrite, --force, --replace                                                                                                                                                                                                                                                                                                                                                                                                 │ bool │                  │ ✓       │ MPM_OVERWRITE           │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.backup.toml_path       │ cloup._params.Argument                 │ [TOML_PATH]                                                                                                                                                                                                                                                                                                                                                                                                                     │ str  │                  │ ✓       │                         │ -                                                        │ -                                                        │ DEFAULT     │
│ mpm.backup.update_version  │ cloup._params.Option                   │ --update-version                                                                                                                                                                                                                                                                                                                                                                                                                │ bool │                  │ ✓       │ MPM_UPDATE_VERSION      │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.install.packages_specs │ cloup._params.Argument                 │ PACKAGES_SPECS...                                                                                                                                                                                                                                                                                                                                                                                                               │ list │                  │ ✓       │                         │                                                          │                                                          │ DEFAULT     │
│ mpm.installed.duplicates   │ cloup._params.Option                   │ -d, --duplicates                                                                                                                                                                                                                                                                                                                                                                                                                │ bool │                  │ ✓       │ MPM_DUPLICATES          │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.outdated.plugin_output │ cloup._params.Option                   │ --plugin-output                                                                                                                                                                                                                                                                                                                                                                                                                 │ bool │                  │ ✓       │ MPM_PLUGIN_OUTPUT       │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.remove.packages_specs  │ cloup._params.Argument                 │ PACKAGES_SPECS...                                                                                                                                                                                                                                                                                                                                                                                                               │ list │                  │ ✓       │                         │                                                          │                                                          │ DEFAULT     │
│ mpm.restore.toml_files     │ cloup._params.Argument                 │ TOML_FILES...                                                                                                                                                                                                                                                                                                                                                                                                                   │ list │                  │ ✓       │                         │                                                          │                                                          │ DEFAULT     │
│ mpm.search.exact           │ cloup._params.Option                   │ --exact / --fuzzy                                                                                                                                                                                                                                                                                                                                                                                                               │ bool │                  │ ✓       │ MPM_EXACT               │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.search.extended        │ cloup._params.Option                   │ --extended / --id-name-only                                                                                                                                                                                                                                                                                                                                                                                                     │ bool │                  │ ✓       │ MPM_EXTENDED            │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.search.query           │ cloup._params.Argument                 │ QUERY                                                                                                                                                                                                                                                                                                                                                                                                                           │ str  │                  │ ✓       │                         │                                                          │                                                          │ DEFAULT     │
│ mpm.search.refilter        │ cloup._params.Option                   │ --refilter / --no-refilter                                                                                                                                                                                                                                                                                                                                                                                                      │ bool │                  │ ✓       │ MPM_REFILTER            │ True                                                     │ True                                                     │ DEFAULT     │
│ mpm.upgrade.all            │ cloup._params.Option                   │ -A, --all                                                                                                                                                                                                                                                                                                                                                                                                                       │ bool │                  │ ✓       │ MPM_ALL                 │ False                                                    │ False                                                    │ DEFAULT     │
│ mpm.upgrade.packages_specs │ cloup._params.Argument                 │ [PACKAGES_SPECS]...                                                                                                                                                                                                                                                                                                                                                                                                             │ list │                  │ ✓       │                         │                                                          │                                                          │ DEFAULT     │
╰────────────────────────────┴────────────────────────────────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴──────┴──────────────────┴─────────┴─────────────────────────┴──────────────────────────────────────────────────────────┴──────────────────────────────────────────────────────────┴─────────────╯
```


### List package managers

```shell-session
╭────────────┬────────────────────┬────────────────┬─────────────────────────────────────┬────────────┬──────────────╮
│ Manager ID │ Name               │ Supported      │ CLI                                 │ Executable │ Version      │
├────────────┼────────────────────┼────────────────┼─────────────────────────────────────┼────────────┼──────────────┤
│ apm        │ Atom's apm         │ ✓ (deprecated) │ ✘ apm not found                     │            │              │
│ apt        │ APT                │ ✓              │ ✓ /usr/bin/apt                      │ ✓          │ ✓ 2.4.8      │
│ apt-mint   │ Linux Mint's apt   │ ✓              │ ✓ /usr/bin/apt                      │ ✓          │ ✘            │
│ brew       │ Homebrew Formulae  │ ✓              │ ✘ brew not found                    │            │              │
│ cargo      │ Rust's cargo       │ ✓              │ ✓ /usr/bin/cargo                    │ ✓          │ ✓ 1.65.0     │
│ cask       │ Homebrew Cask      │ ✘ macOS only   │ ✘ brew not found                    │            │              │
│ choco      │ Chocolatey         │ ✘ Windows only │ ✘ choco not found                   │            │              │
│ composer   │ PHP's Composer     │ ✓              │ ✘ composer not found                │            │              │
│ dnf        │ DNF                │ ✓              │ ✘ dnf not found                     │            │              │
│ emerge     │ Emerge             │ ✓              │ ✘ emerge not found                  │            │              │
│ flatpak    │ Flatpak            │ ✓              │ ✘ flatpak not found                 │            │              │
│ gem        │ Ruby Gems          │ ✓              │ ✓ /usr/bin/gem                      │ ✓          │ ✓ 3.3.5      │
│ mas        │ Mac AppStore       │ ✘ macOS only   │ ✘ mas not found                     │            │              │
│ npm        │ Node's npm         │ ✓              │ ✓ /home/paxsali/.opt/nodejs/bin/npm │ ✓          │ ✓ 9.2.0      │
│ opkg       │ OPKG               │ ✓              │ ✘ opkg not found                    │            │              │
│ pacaur     │ Pacaur             │ ✓              │ ✘ pacaur not found                  │            │              │
│ pacman     │ Pacman             │ ✓              │ ✘ pacman not found                  │            │              │
│ paru       │ Paru               │ ✓              │ ✘ paru not found                    │            │              │
│ pip        │ Pip                │ ✓              │ ✓ /usr/bin/python3                  │ ✓          │ ✓ 23.0.1     │
│ pipx       │ Pipx               │ ✓              │ ✓ /home/paxsali/.local/bin/pipx     │ ✓          │ ✓ 1.2.0      │
│ scoop      │ Scoop              │ ✘ Windows only │ ✘ scoop not found                   │            │              │
│ snap       │ Snap               │ ✓              │ ✓ /usr/bin/snap                     │ ✓          │ ✓ 2.58+22.04 │
│ steamcmd   │ Valve Steam        │ ✓              │ ✘ steamcmd not found                │            │              │
│ vscode     │ Visual Studio Code │ ✓              │ ✓ /usr/bin/code                     │ ✓          │ ✓ 1.76.1     │
│ yarn       │ Node's yarn        │ ✓              │ ✘ yarn not found                    │            │              │
│ yay        │ Yay                │ ✓              │ ✘ yay not found                     │            │              │
│ yum        │ YUM                │ ✓              │ ✘ yum not found                     │            │              │
│ zypper     │ Zypper             │ ✓              │ ✘ zypper not found                  │            │              │
╰────────────┴────────────────────┴────────────────┴─────────────────────────────────────┴────────────┴──────────────╯
```


### Meta Package Manager version

```shell-session
mpm, version 5.12.0
```

**Comment by kdeldycke** [comment]
Thanks @paxsali for the bug report and yes, I confirm your analysis.

I hard-coded a check on `sudo` usage a while back to prevent its call on non-supported platforms, but I naively restricted it to Linux. Now that I can properly identify all platforms, including the macOS, *BSD and other UNIXes, I was able to fix this issue in: 479c9f38e7150264ce5a950e635c334c608820a6.

The fix will be available in the upcoming release of mpm `5.13.0`.

**Comment by kdeldycke** [comment]
FYI meta-package-manager v5.13.0 has just been released.
